roofit/roofitcore/src/RooWorkspace.cxx

```diff
@@ -1481,6 +1481,25 @@ std::list<TObject*> RooWorkspace::allGenericObjects() const
 }
 
 
+namespace {
+
+std::string findFileInPath(std::string const &file, std::list<std::string> const &dirList)
+{
+   // Check list of additional paths
+   for (std::string const &diter : dirList) {
+
+      char *cpath = gSystem->ConcatFileName(diter.c_str(), file.c_str());
```

Comment:
This `ConcatFileName` function is horrible: any chance we could deprecate it and provide a new one that returns a `std::string`? (Looking at its implementation it doesn't even make sense that it returns a `char *` rather than a `TString`...)

Reply:
Of course it is horrible, haha! But can you maybe still approve the fix I have now, so I can merge and backport it? For the future, I agree we should absolutely get rid of this function.

Reply:
Yes, sorry, I wasn't implying that it should be done in this PR, just something to think of for the future :)